cumulusci/utils/http/multi_request.py

```diff
@@ -100,13 +127,62 @@ def __enter__(self, *args):
     def __exit__(self, *args):
         self.close()
 
-    def do_composite_requests(self, requests):
+    def do_composite_requests(
+        self, requests
+    ) -> T.Tuple[T.Sequence, T.Sequence]:  # results, erros
         if not self.psf:
             raise AssertionError(
                 "Session was not opened. Please call open() or use as a context manager"
             )
 
         composite_requests = create_composite_requests(requests, self.chunk_size)
-        composite_results = self.psf.do_requests(composite_requests)
+        composite_results, errors = self.psf.do_requests(composite_requests)
         individual_results = parse_composite_results(composite_results)
-        return list(individual_results)
+
+        errors = list(errors)
+        if errors:
+            singleton_results, unrecoverable_errors = self.retry_errors(errors)
+        else:
+            singleton_results = []
+            unrecoverable_errors = []
+
+        individual_results = list(individual_results)
+
+        individual_results += list(singleton_results)
+        return individual_results, unrecoverable_errors
+
+    def retry_errors(self, errors):
```

Comment:
docstring

Reply:
Why do you prefer namedtuples over dataclasses?

Reply:
We haven't removed support for 3.6 yet, nor added a dataclasses backport library, as far as I know.

Reply:
Right, makes sense. Should we consider switching to dataclasses once we've dropped 3.6?

Reply:
We could have a period of experimenting with DataClasses and seeing whether there are any downsides.

We might discover that there are some cases where we might still like NamedTuples for minor reasons:

1. Immutability can make reasoning slightly easier.
2. We often import "T" and therefore T.NamedTuple can be used with one less import
3. In the Snowfakery context (and perhaps other bulkdata contexts) NamedTuples would probably be less memory and perhaps other perf benefits? 

I mean if you want mutability or the tons of other features of Dataclasses then of course they are better. But if you "would" have used a tuple before either of them existed (simple pairs and triples) and you just want to use names instead of positions, I don't see a harm with using NamedTuples. The forward migration path from NamedTuples to Dataclasses is probably pretty trivial in almost every case, so if your model evolves towards more complexity it isn't hard to switch. But I'm certainly not a NamedTuple zealot and I might find that there's some detail of dataclasses that is so much better that makes them better by default.